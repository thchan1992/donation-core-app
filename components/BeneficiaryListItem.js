import { TouchableOpacity, View, Text, StyleSheet, Image } from "react-native";

const BeneficiaryListItem = ({ name, imageUrl, onPress }) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Image
        style={{
          height: 150,
          width: 200,
          alignSelf: "center",
          justifyContent: "center",
          borderRadius: 8,
          borderWidth: 1.5,
          borderColor: "#FF304F",
        }}
        source={imageUrl}
      />
      <Text style={styles.text}>{name}</Text>
    </TouchableOpacity>
  );
};

export default BeneficiaryListItem;

const styles = StyleSheet.create({
  button: {
    height: 180,
    width: 200,
    margin: 5,
    borderRadius: 8,
    backgroundColor: "#28C7FA",
    borderRadius: 8,
    borderWidth: 1.5,
    borderColor: "#FF304F",
  },
  text: {
    alignSelf: "center",
    padding: 2.5,
    fontWeight: "bold",
    textTransform: "uppercase",
    fontSize: 20,
    color: "black",
  },
});
