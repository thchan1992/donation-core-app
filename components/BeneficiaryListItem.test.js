import renderer from "react-test-renderer";
import React from "react";
import BeneficiaryListItem from "./BeneficiaryListItem";
import { TouchableOpacity } from "react-native";

describe("BeneficiaryListItem", () => {
  let onPress;

  beforeEach(() => {
    onPress = jest.fn();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BeneficiaryListItem
          name="Test Charity"
          imageUrl={require("../assets/adaptive-icon.png")}
          onPress={onPress}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onPress when pressed", () => {
    const component = renderer.create(
      <BeneficiaryListItem
        name="Test Charity"
        imageUrl={require("../assets/adaptive-icon.png")}
        onPress={onPress}
      />
    );
    const instance = component.root;
    const button = instance.findByType(TouchableOpacity);
    button.props.onPress();
    expect(onPress).toHaveBeenCalled();
  });
});
