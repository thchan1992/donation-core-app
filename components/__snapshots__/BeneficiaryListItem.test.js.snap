// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeneficiaryListItem renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#28C7FA",
      "borderColor": "#FF304F",
      "borderRadius": 8,
      "borderWidth": 1.5,
      "height": 180,
      "margin": 5,
      "opacity": 1,
      "width": 200,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "#FF304F",
        "borderRadius": 8,
        "borderWidth": 1.5,
        "height": 150,
        "justifyContent": "center",
        "width": 200,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "black",
        "fontSize": 20,
        "fontWeight": "bold",
        "padding": 2.5,
        "textTransform": "uppercase",
      }
    }
  >
    Test Charity
  </Text>
</View>
`;
